executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Header.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+
+module Cardano.Ledger.Spec.STS.Chain.Header where
+
+import           GHC.Generics (Generic)
+--import           Data.Monoid.Generic (GenericMonoid (GenericMonoid),
+--                     GenericSemigroup (GenericSemigroup))
+import           Data.Typeable (typeOf)
+import qualified Data.Map.Strict as Map
+
+import           Cardano.Crypto.Hash (Hash, HashAlgorithm)
+
+import           Control.State.Transition (Embed, Environment, PredicateFailure,
+                     STS, Signal, State, TRC (TRC), IRC (IRC), initialRules,
+                     judgmentContext, trans, transitionRules, wrapFailed, (?!))
+import           Data.AbstractSize (HasTypeReps)
+
+import           Ledger.Core (Slot)
+
+import           Cardano.Ledger.Spec.STS.Update.Hupdate (HUPDATE)
+import qualified Cardano.Ledger.Spec.STS.Update.Hupdate as Hupdate
+import           Cardano.Ledger.Spec.STS.Sized (Sized, costsList)
+import           Cardano.Ledger.Spec.STS.Update.Data (SIPData)
+
+
+-- | The Block HEADER STS
+data HEADER hashAlgo
+
+data Env hashAlgo
+  = Env { initialSlot :: !Slot
```

Comment:
Do we need the `initialSlot` here at all? As far as I can see we use `initiaSlot` in the CHAIN rules to bootstrap the current slot. But I don't think we need this information in the children.